test/plugin/smartPay.nettest.ts

```diff
@@ -0,0 +1,119 @@
+import { assert } from 'chai'
+import {
+  EdgeCorePluginOptions,
+  EdgeCurrencyPlugin,
+  EdgeCurrencyTools,
+  makeFakeIo
+} from 'edge-core-js'
+import { before, describe, it } from 'mocha'
+import fetch from 'node-fetch'
+
+import { objectCheckOneWay } from '../../src/common/utils'
+import edgeCorePlugins from '../../src/index'
+import { fakeLog } from '../fake/fakeLog'
+
+const smartPayPublicAddress = 'TUmgPbM5J6om7Z2PJjzrbSEbXit84ZhVCj'
+
+type SmartPayTestFixture = Array<{
+  WALLET_TYPE: string
+  pluginId: string
+  currencyCode: string
+  uri: string
+  parsedUri: any
+}>
+
+const fixtures: SmartPayTestFixture = [
+  {
+    WALLET_TYPE: 'wallet:tron',
+    pluginId: 'tron',
+    currencyCode: 'TRON',
+    uri: '00020126360014BR.GOV.BCB.PIX0114+5548988356069520400005303986540511.115802BR5925Francisco Rocelo Bezerra 6009SAO PAULO61080540900062070503***630464C2',
+    parsedUri: {
+      currencyCode: 'USDT',
+      metadata: {
+        name: 'Francisco Rocelo Bezerra ',
+        notes: `To PIX: +5548988356069`
+      },
+      publicAddress: smartPayPublicAddress,
+      uniqueIdentifier:
+        '00020126360014BR.GOV.BCB.PIX0114+5548988356069520400005303986540511.115802BR5925Francisco Rocelo Bezerra 6009SAO PAULO61080540900062070503***630464C2'
+    }
+  },
+  {
+    WALLET_TYPE: 'wallet:tron',
+    pluginId: 'tron',
+    currencyCode: 'TRON',
+    uri: 'rocelo@smartpay.com.vc',
+    parsedUri: {
+      currencyCode: 'USDT',
+      metadata: {
+        name: 'PIX: rocelo@smartpay.com.vc',
+        notes: `To PIX: rocelo@smartpay.com.vc`
+      },
+      publicAddress: smartPayPublicAddress,
+      uniqueIdentifier: 'rocelo@smartpay.com.vc'
+    }
+  },
+  {
+    WALLET_TYPE: 'wallet:tron',
+    pluginId: 'tron',
+    currencyCode: 'TRON',
+    uri: '+55 48 98835 6069',
+    parsedUri: {
+      currencyCode: 'USDT',
+      metadata: {
+        name: 'PIX: +5548988356069',
+        notes: `To PIX: +5548988356069`
+      },
+      publicAddress: smartPayPublicAddress,
+      uniqueIdentifier: '+5548988356069'
+    }
+  },
+  {
+    WALLET_TYPE: 'wallet:tron',
+    pluginId: 'tron',
+    currencyCode: 'TRON',
+    uri: '1700830b-fbc5-4bd1-b714-ee83b3b0da26',
+    parsedUri: {
+      currencyCode: 'USDT',
+      metadata: {
+        name: 'PIX: 1700830b-fbc5-4bd1-b714-ee83b3b0da26',
+        notes: `To PIX: 1700830b-fbc5-4bd1-b714-ee83b3b0da26`
+      },
+      publicAddress: smartPayPublicAddress,
+      uniqueIdentifier: '1700830b-fbc5-4bd1-b714-ee83b3b0da26'
+    }
+  }
+]
+
+for (const fixture of fixtures) {
+  let tools: EdgeCurrencyTools
+
+  const WALLET_TYPE = fixture.WALLET_TYPE
+
+  const fakeIo = makeFakeIo()
```

Comment:
Can you put this with the normal tests? There is no reason to create a whole new "nettest" command, unless we were maybe trying to separate tests that hit the network from tests that run locally. In that case, maybe we could go deeper on the separation, or at least create a task to do it later.

If you don't want to run the whole test suite during development, you can just do `mocha -g <seach term>` to just run matching tests.

Then, since this test seems to focus on SmartPay, maybe use this testing setup:

```ts
describe('SmartPay', async function () {
   // The same setup logic you already have:
   const fakeIo = ...
   const plugin = ...
   const tools = await plugin.getCurrencyTools()

  // The same loop you already have, but moved inside the `describe`:
  for (const fixture of fixtures) {
    it(fixture.uri, function() {
       // The same test as before...
     })
  })
})
```

This is more idiomatic. The `describe` block is meant to group related tests together, so it needs to be on the outside. Then, since `describe` makes it possible to use `async` / `await`, you can ditch the ugly `before` block.

Assuming the "nettest" separation really is about network-using tests, then this would ***not*** be a place to test generic URI parsing features. Therefore, this test should be named "SmartPay", specifically, so we aren't tempted to recycle it.